src/operations.jl

```diff
@@ -5,25 +5,11 @@ hash(c::AbstractRGB, hx::UInt) = hash(blue(c), hash(green(c), hash(red(c), hx)))
 hash(c::TransparentRGB, hx::UInt) = hash(alpha(c), hash(blue(c), hash(green(c), hash(red(c), hx))))
 
 # rand
-typealias RandTypes{T,C} Union{AbstractRGB{T},
-                               TransparentRGB{C,T},
-                               AbstractGray{T},
-                               TransparentGray{C,T}}
-typealias RandTypesFloat{C,T<:AbstractFloat} RandTypes{T,C}
-
-rand{G<:AbstractGray}(::Type{G}) = G(rand())
-rand{G<:TransparentGray}(::Type{G}) = G(rand(), rand())
-rand{C<:AbstractRGB}(::Type{C}) = C(rand(), rand(), rand())
-rand{C<:TransparentRGB}(::Type{C}) = C(rand(), rand(), rand(), rand())
-function rand{C<:RandTypesFloat}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(eltype(C), (sizeof(C)÷sizeof(eltype(C)), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f8}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt8, (sizeof(C), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f16}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt16, (sizeof(C)>>1, sz...)), sz)
-end
+_rand{T<:Colorant}(::Type{T}) = mapc(x->rand(eltype(T)), base_colorant_type(T)())
+_rand{T<:Colorant}(::Type{T}, sz::Dims) =
+      reinterpret(T, rand(eltype(T), (div(sizeof(T), sizeof(eltype(T))), sz...)), sz)
+rand{T<:Colorant}(::Type{T}, sz::Dims...) = _rand(ccolor(T, base_colorant_type(T){Float64}), sz...)
+rand{T<:HSV}(::Type{T}, sz::Dims...) = error("rand(::Type{HSV}, sz...) not supported")
```

Comment:
Sorry, I was just using HSV as an example. You'll have problems with lots of others too. For example:
```julia
julia> convert(DIN99d, rand(RGB))
DIN99d{Float64}(17.531226651912203,-1.7876008277942326,-27.780870734776467)
```
that single example goes way beyond the bounds of `[0,1]`. The fundamental question is, can `rand` sample the "gamut" of the colorspace? Even `LMS` and `XYZ` extend higher than 1.